theano/tensor/elemwise.py

```diff
@@ -1059,35 +1076,110 @@ def _c_all(self, node, nodename, inames, onames, sub):
                                      fortran=alloc_fortran)
             alloc += cgen.make_checks([list(range(nnested))], [odtype],
                                       dict(sub, lv0=oname))
-        olv_index = i  # index of the last output
+        # index of the last output
```

Comment:
All the inplace work is done after the alloc of the outputs was done. So mostly, we alloc an output, then we free it. Why not reverse this, set reference of outputs that are destroy of inputs, then alloc the others?

The call make_alloc will only allocate the output if the isn't NULL. In that case, he call resize on it that will do nothing as it will be of the good size.